Lesson34/gozde/lesson34-app/src/UserContext.jsx

```diff
@@ -0,0 +1,39 @@
+import { createContext, useReducer, useState } from 'react';
+
+
+// Step 1 - create context
+export const UserContext = createContext();
+export const UserDispatchContext = createContext();
+
+export const UserProvider = ({children, initialState}) => {
+       const [user, dispatch] = useReducer(userReducer, initialState ?? {});
+
+
+    return (
+        //step 2 - wrap application in context provider
+		<UserContext.Provider value={loggedOutUser}>
+            <UserDispatchContext.Provider value={dispatch}>
+               {children}
+            </UserDispatchContext.Provider>
+        </UserContext.Provider>
+    )
+};
+
+export const UserActionTypes = {
+    login: 'login',
+    logout: 'logout',
+    update: 'update'
+};
+
+function userReducer (state, action){
+switch(action.type) {
+    case 'login':
```

Comment:
Since we have an object set up for holding different action types, use it in the switch case:
```suggestion
    case UserActionTypes.login:
```

This approach helps avoiding mistakes when typing the code.